### PR TITLE
Initialize SR-IOV devices and resource pool

### PR DESCRIPTION
**Pull request by lvfxx**
Add methods for initialising SR-IOV devices and populating resource pool

### CODE REVIEW

File: pkg/sriov/utils/sriov_provider.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ import (
 )
 
 const (
+	// SysfsDevicesPath is path to the devices info in Linux sysfs pseudo file system
+	SysfsDevicesPath     = "/sys/bus/pci/devices/"
```
Comment (denis-tingajkin) [inline pkg/sriov/utils/sriov_provider.go:38]: Can we make this as private const for types?

File: pkg/sriov/types.go [hunk 2]
```diff
@@ -114,6 +120,108 @@ func (n *NetResourcePool) GetFreeVirtualFunctionsInfo() *FreeVirtualFunctionsInf
...
+
+		pfIfaceNames, err := sriovProvider.GetNetInterfacesNames(ctx, pfPciAddr)
+		if err != nil {
+			return errors.Wrapf(err, "unable to determine net interface name for device %s", pfPciAddr)
+		}
+		if len(pfIfaceNames) != 1 {
+			return errors.Errorf("expected 1 network interface name, actual: %d", len(pfIfaceNames))
+		}
+
+		physfun := &PhysicalFunction{
```
[the remark this answers is not in the document]
Comment (lvfxx) [reply pkg/sriov/types.go]: Yes, agree with you. Also we need rename `TargetMacAddress` to `TargetPCIAddress` in config

File: pkg/sriov/utils/sriov_provider.go (hunk 1 above)
Comment (lvfxx) [reply pkg/sriov/utils/sriov_provider.go]: Yes, we can. God idea :)

File: pkg/sriov/types.go [hunk 3]
```diff
@@ -114,6 +120,108 @@ func (n *NetResourcePool) GetFreeVirtualFunctionsInfo() *FreeVirtualFunctionsInf
...
+	if sriovProvider.IsSriovConfigured(ctx, pciAddr) {
+		return errors.Errorf("device %s is already configured", pciAddr)
+	}
+
+	// exclude net device in-use in host
+	ifaceNames, err := sriovProvider.GetNetInterfacesNames(ctx, pciAddr)
+	if err != nil {
+		return errors.Wrapf(err, "unable to determine net interface name for device %s", pciAddr)
+	}
+	if isDefaultRoute := isDefaultRoute(ifaceNames); isDefaultRoute {
```
[the remark this answers is not in the document]
Comment (lvfxx) [reply pkg/sriov/types.go]: I think there is no problem, if we don't interact with physical function, i'll remove this check

**Pull request merged**